### ISSUE TITLE
floating ip race condition in openstack node driver

### ISSUE THREAD

**Issue by SvenMW** [rancher/rancher#27864]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
configure latest rancher to use openstack node driver  
deploy a kubernetes cluster with floating ips with vms from openstack (train release), at least 3 nodes (more is better)

**Result:**
watch in parallel in openstack and rancher how floating ips get assigned:
rancher allocates (in rancher gui) the same floating ip to different nodes
this obviously doesn't work in openstack, the floating ip never gets assigned to one of the nodes

**Other details that may be helpful:**
already reported here: https://github.com/docker/machine/issues/4038

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.4.5
- Installation option (single install/HA): single install, in the docker env (just followed the setup guide), for poc purpose

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): infra provider is openstack
- Machine type (cloud/VM/metal) and specifications (CPU/memory): openstack vms
- Kubernetes version (use `kubectl version`): doesn't matter

```
(paste the output here)
```

- Docker version (use `docker version`): doesn't matter

```
(paste the output here)
```

HTH

if you need any other info, see the linked bug report, or please ask :)

**Comment by wby1089** [comment]
I experienced the same issue with Openstack Train release. I hope this issue will be resolved as soon as possible.

**Comment by janeczku** [comment]
Issue is reproduced on Rancher 2.5.9 and RH Openstack 16.

**Comment by janeczku** [comment]
Fix has been submitted to rancher/machine repository: https://github.com/rancher/machine/pull/145

**Comment by thedadams** [comment]
The proposed fix has been merged into rancher/machine and the new version of rancher/machine has been included into Rancher.

I am not sure if QA has the ability to test these changes, but I will move the issue to "To Test" for now.

**Comment by Jono-SUSE-Rancher** [comment]
Hey @janeczku since you've got some repro steps above, do you think you would be able to try and validate this issue and close it if satisfied? Our QA team doesn't easily have OpenStack creds that can be used in conjunction with validation of this issue.

@thedadams has merged his fix into rancher/machine and it should be safe to validate.

**Comment by Jono-SUSE-Rancher** [comment]
Hello @SvenMW - since you were the original reporter and we don't have OpenStack creds, would you be able to do some validation/verification of this fix? You could try the [v2.6.1 RC](https://github.com/rancher/rancher/releases/tag/v2.6.1-rc1) for validation.

**Comment by SvenMW** [comment]
I'll try to get some time for this, but I would appreciate if someone else could also test this as we are currently not really using rancher, because of the apparently lacking support of openstack based deployments (I hoped that would change since Suse acquired Rancher, but I digress).

Will try, but can't promise! Thanks for all the work on this anyway.

**Comment by Jono-SUSE-Rancher** [comment]
Thanks @SvenMW - understood. We've spent some cycles on getting it working but don't have an efficient way to test it. We can include it but release note that it might a bit untested. @sowmyav27 - not sure how you want to proceed in terms of the life of the ticket itself.

**Comment by janeczku** [comment]
@Jono-SUSE-Rancher I can confirm that the issue is no longer present in v2.6.1-rc1 (validated on RH Openstack 16.1.x)

**Comment by Jono-SUSE-Rancher** [comment]
Thanks for confirmation @janeczku ! 
It's all yours now @sowmyav27 - do as you please (close/move/etc).
